sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -228,14 +228,21 @@ object SharedState extends Logging {
       sparkConf: SparkConf,
       hadoopConf: Configuration,
       initialConfigs: scala.collection.Map[String, String] = Map.empty): Unit = {
+
+    def containsInSparkConf(key: String): Boolean = {
+      sparkConf.contains(key) || sparkConf.contains("spark.hadoop." + key) ||
+        (key.startsWith("hive") && sparkConf.contains("spark." + key))
```

Comment:
nit: It seems there is no test for the condition `(key.startsWith("hive") && sparkConf.contains("spark." + key))`?